test/over_react/component_declaration/transformer_integration_tests/required_accessor_integration_test.dart

```diff
@@ -0,0 +1,161 @@
+        );
+      });
+    });
+
+    group('throwing when a prop is nullable and not set', () {
+      test('on mount', () {
+        expect(() => render(ComponentTest()..requiredProp = true),
+            throwsPropError_Required('ComponentTestProps.nullableProp'));
+      });
+
+      test('on re-render', () {
+        var mountNode = new DivElement();
+        react_dom.render((ComponentTest()
+          ..requiredProp = true
+          ..nullableProp = true
+        )(), mountNode);
+
+        expect(() => react_dom.render((ComponentTest()..requiredProp = true)(), mountNode),
+            throwsPropError_Required('ComponentTestProps.nullableProp', 'This prop can be set to null!')
+        );
+      });
+    });
+
+    group('not throwing when a prop is required and set', () {
+      test('on mount', () {
+        expect(() => render(ComponentTest()
+          ..nullableProp = true
+          ..requiredProp = true
+        ), returnsNormally);
+      });
+
+      test('on re-render', () {
+        var mountNode = new DivElement();
+        react_dom.render((ComponentTest()
+          ..requiredProp = true
+          ..nullableProp = true
+        )(), mountNode);
+
+        expect(() => react_dom.render((ComponentTest()
+          ..requiredProp = true
+          ..nullableProp = true
+        )(), mountNode), returnsNormally);
+      });
+    });
+
+    group('not throwing when a prop is nullable and set to null', () {
+      test('on mount', () {
+        expect(() => render(ComponentTest()
+          ..nullableProp = null
+          ..requiredProp = true
+        ), returnsNormally);
+      });
+
+      test('on re-render', () {
+        var mountNode = new DivElement();
+        react_dom.render((ComponentTest()
+          ..requiredProp = true
+          ..nullableProp = true
+        )(), mountNode);
+
+        expect(() => react_dom.render((ComponentTest()
+          ..requiredProp = true
+          ..nullableProp = null
+        )(), mountNode), returnsNormally);
+      });
+    });
+  });
+
+  test('requiredProp shorthands `const Accessor(isRequired: true)`', () {
+    expect(requiredProp, const Accessor(isRequired: true));
+  });
+
+  test('nullableRequiredProp shorthands `const Accessor(isRequired: true, isNullable: true)`', () {
+    expect(nullableRequiredProp, const Accessor(isRequired: true, isNullable: true));
+  });
+}
+
+@Factory()
+UiFactory<ComponentTestProps> ComponentTest;
+
+@Props()
+class ComponentTestProps extends UiProps {
+  @Accessor(isRequired: true, requiredErrorMessage: 'This Prop is Required for testing purposes.')
+  var requiredProp;
+
+  @Accessor(isRequired: true, isNullable: true, requiredErrorMessage: 'This prop can be set to null!')
+  var nullableProp;
+}
+
+@Component()
+class ComponentTestComponent extends UiComponent<ComponentTestProps> {
+  @override
+  render() => Dom.div()();
+}
```

Comment:
We should have integration tests for the alias/shorthand consts as well.

I actually don't think that they're getting picked up right now...